tox.ini

```diff
@@ -1,29 +1,122 @@
+base = testenv, test
+description = Run integration tests with pytest
+labels =
+    py38, py310, py311: tests, integration-tests
+commands = pytest {tty:--color=yes} --junit-xml=results/test-results-{env_name}.xml tests/integration {posargs}
+
+[lint]  # Standard linting configuration
+package = editable
+extras =
+    dev
+    test
+env_dir = {work_dir}/linting
+runner = ignore_env_name_mismatch
+
+[shellcheck]
+find = git ls-files
+filter = file --mime-type -Nnf- | grep shellscript | cut -f1 -d:
+
+[testenv:lint-{black,ruff,shellcheck,codespell,yaml}]
+description = Lint the source code
+base = testenv, lint
+labels = lint
+allowlist_externals =
+    shellcheck: bash, xargs
+commands_pre =
+    shellcheck: bash -c '{[shellcheck]find} | {[shellcheck]filter} > {env_tmp_dir}/shellcheck_files'
 commands =
-    pip install -U pip
-    pytest --basetemp={envtmpdir}
+    black: black --check --diff {tty:--color} {posargs} .
+    ruff: ruff --diff --respect-gitignore {posargs} .
+    shellcheck: xargs -ra {env_tmp_dir}/shellcheck_files shellcheck
+    codespell: codespell --toml {tox_root}/pyproject.toml {posargs}
+    yaml: yamllint {posargs} .
 
-[testenv:docs]
-commands = make docs
+[testenv:lint-{mypy,pyright}]
+description = Static type checking
+base = testenv, lint
+env_dir = {work_dir}/typing
+extras = dev, types
+labels = lint, type
+allowlist_externals =
+    mypy: mkdir
+commands_pre =
+    mypy: mkdir -p .mypy_cache
+commands =
+    pyright: pyright {posargs}
+    mypy: mypy --install-types --non-interactive {posargs:.}
+
+[testenv:format-{black,ruff,codespell}]
+description = Automatically format source code
+base = testenv, lint
+labels = format
+commands =
+    black: black {tty:--color} {posargs} .
+    ruff: ruff --fix --respect-gitignore {posargs} .
+    codespell: codespell --toml {tox_root}/pyproject.toml --write-changes {posargs}
 
-[testenv:lint]
-commands = make lint
+[docs]  # Sphinx documentation configuration
+extras = doc
+package = editable
+env_dir = {work_dir}/docs
+runner = ignore_env_name_mismatch
 
-[testenv:units]
-commands = make test-units
+[testenv:build-docs]
+description = Build sphinx documentation
+base = docs
+allowlist_externals = bash
+commands_pre = bash -c 'if [[ ! -e docs ]];then echo "No docs directory. Run `tox run -e sphinx-quickstart` to create one.;";return 1;fi'
+# "-W" is to treat warnings as errors
+commands = sphinx-build {posargs:-b html} -W {tox_root}/docs {tox_root}/docs/_build
 
-[testenv:integrations]
-commands = make test-integrations
+[testenv:autobuild-docs]
+description = Build documentation with an autoupdating server
+base = docs
+commands = sphinx-autobuild {posargs:-b html --open-browser --port 8080} -W --watch {tox_root}/craft-parts {tox_root}/docs {tox_root}/docs/_build
 
+[testenv:lint-docs]
+description = Lint the documentation with sphinx-lint
+base = docs
+commands = sphinx-lint --ignore docs/_build --enable=all --disable=line-too-long,trailing-whitespace {posargs} docs/
```

Comment:
Existing documentation has long lines and trailing whitespace, so I'm disabling those checks